store/conf/attrs/zimbra-attrs.xml

```diff
@@ -9925,8 +9925,66 @@ TODO: delete them permanently from here
   <defaultCOSValue>FALSE</defaultCOSValue>
   <desc>Creates unsubscribe system folder</desc>
  </attr>
- <attr id="4004" name="zimbraPrefCalenderScaling" type="enum" value="10,15,30" cardinality="single" optionalIn="account,cos" flags="accountInherited,domainAdminModifiable" since="9.1.0">
+<attr id="4004" name="zimbraPrefCalenderScaling" type="enum" value="10,15,30" cardinality="single" optionalIn="account,cos" flags="accountInherited,domainAdminModifiable" since="9.1.0">
   <defaultCOSValue>30</defaultCOSValue>
   <desc>Default calendar resolution preference</desc>
 </attr>
+
+<attr id="4007" name="zimbraBackupConfigEnabled" type="boolean" cardinality="single" optionalIn="globalConfig,domain,cos,account" flags="accountCosDomainInherited,domainInherited" since="9.1.0">
+  <desc>Whether or not account is eligible for backup
+        If true on cos level then backup accounts for cos. zimbraDomainDefaultCOSId is considered.
+        If unset on cos level but true on domain level then backup domains account.
+        If unset on cos level, unset on domain level but true on account level then specific account will be backed up.
+        If true on cos level , unset on domain level but zimbraDomainDefaultCOSId is set to cos then accounts for that domain will be backed up.
+  </desc>
+</attr>
+
+<attr id="4008" name="zimbraBackupEnabled" type="boolean" cardinality="single" optionalIn="globalConfig" since="9.1.0">
+  <globalConfigValue>TRUE</globalConfigValue>
+  <desc>If true, enables backup globally and if there is pre-configured data the crontab will get that schedule.</desc>
+</attr>
+
+<attr id="4009" name="zimbraBackupSyncUsersEnabled" type="boolean" cardinality="single" optionalIn="globalConfig" since="9.1.0">
+  <globalConfigValue>FALSE</globalConfigValue>
+  <desc>If true, will synchronize dumpster enabled accounts with backup eligible accounts from zimbraBackupConfigEnabled list.</desc>
+</attr>
+
+<attr id="4010" name="zimbraBackupStartTime" type="string" max="64" cardinality="single" optionalIn="globalConfig,server" flags="serverInherited" since="9.1.0">
```

Comment:
From UI perspective we are giving facility to set day and time only to admins for now, but having ldap attributes like day and time would not be scalable in future
like in future what if I want to provide facility to admin to set more fine grain control on this like below
- run full back on monday and thursday every week
- run full backup on alternate weeks on every sunday
- run full backup on monthly basis
Can't we have an ldap attribute where you can specify full cron compatible string, which will be used as it is when updating crontab
Same cron entry can be used also when we remove usage of crontab and convert to service

as I am asking this as a requirement change, I am ok to fork this into separate ticket if that is preferable

Reply:
@shardool-gore @k-kato @zimsuchitgupta @rcyarrapothu  your thoughts?

Reply:
I agree with you on adding more flexibility to this attribute but it's a string and we can actually pass in different values not just start day or time only. This could be changed to a timestamp easily in future. The logic where we are handling this attribute should be adjusted accordingly. 
And "zimbraBackupStartTime" fits well even if we are passing a full crontab in my opinion.

Reply:
Here is just an idea.
We can define/reserve another attribute to write full crontab setting.
For example,
```
<attr id="xxx" name="zimbraBackupCompleteDefinition" type="string" cardinality="multi" optionalIn="globalConfig,server" flags="serverInherited" since="9.1.0"> 
```
It is `multi`. It will be able to set full backup configuration of crontab like this.
(This is just an example)
```
If these settings want to be set,
   f    0 1 * * 6
   i    0 1 * * 0-5
   d 1m 0 0 * * *

set multiple values:
   zmprov mcf +zimbraBackupCompleteDefinition 'f    0 1 * * 6'
   zmprov mcf +zimbraBackupCompleteDefinition 'i    0 1 * * 0-5'
   zmprov mcf +zimbraBackupCompleteDefinition 'd 1m 0 0 * * *'
```
Need to modify a logic; If it is set, it will overwrites all other backup-related settings, or other settings will be ignored.
If we also need UI, we can add a `textarea` to write the settings.

Reply:
Created new attribute "zimbraBackupCrontabConfig". It's a reserve attribute for future enhancements.